### ISSUE TITLE
Truncated REPL output results in indented REPL prompt

### ISSUE THREAD

**Issue by dgr** [clojure-emacs/cider#2778]
When a large output object is printed in the REPL, if the output is truncated, it leaves the REPL prompt indented in what seems like the pretty-print indentation.

For instance:
```
     #object[org.threeten.extra.Interval 0x719f7581 "2020-02-12T08:00:00Z/2020-02-12T14:15:00Z"]
     #object[org.threeten.extra.Interval 0x1f235d0d "2020-02-12T14:30:00Z/2020-02-12T21:15:00Z"]
     #object[org.threeten.extra.Interval 0x26730ac "2020-02-13T08:00:00Z/2020-02-13T14:15:00Z"]
     #object[org.threeten.extr
             ... output truncated to 1MB ...
             user> 
             user>
             user> 
             user> 
             user> 
             user> 
             user> 
             user> 
             user> (+ 1 2)
             3
             user> 
```

We would expect that the prompt would be reset to the left edge of the buffer. Even restarting the Sesman session doesn't reset.

This is with:
CIDER 0.24.0snapshot (package: 20200109.1720), nREPL 0.6.0
Emacs 26.3 on MacOS, installed from brew.

**Comment by bbatsov** [comment]
That's pretty weird. If I had to guess I'd say that probably that unmatched `[` in the truncated output messed up the indentation logic.

**Comment by dgr** [comment]
I'm thinking that this one might be related to aggressive-indent-mode, which I finally realized that I had enabled globally and not excluded cider-repl-mode. When the REPL output gets truncated, I think aggressive indent mode is trying to do its thing, gets confused because the output is not complete in terms of parenthesis or other identing delimiters, and so it indents something it shouldn't. So, in short, maybe not a CIDER bug at all. I've disabled aggressive-indent-mode for cider-real-mode for now and will monitor how it works and close this bug if I don't see it again shortly.
